src/adapter.ts

```diff
@@ -11,29 +11,36 @@
 
 'use strict';
 
+import { Action } from "./action";
+import { AddonManagerProxy } from "./addon-manager-proxy";
+import { Device } from "./device";
+
 /**
  * Base class for adapters, which manage devices.
  * @class Adapter
  *
  */
-class Adapter {
-  constructor(addonManager, id, packageName, {verbose} = {}) {
-    this.manager = addonManager;
-    this.id = id;
-    this.packageName = packageName;
+export class Adapter {
+  private verbose: boolean;
+  private name = this.constructor.name;
+  private devices: { [key: string]: Device } = {};
```

Comment:
TypeScript has inbuilt `Record<K, T>` so you might use `Record<string, Device>`. Here and in other places.

Reply:
Sounds like a good idea.

Reply:
Done.